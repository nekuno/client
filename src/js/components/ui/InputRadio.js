import React, { PropTypes, Component } from 'react';

export default class InputRadio extends Component {

    static propTypes = {
        value         : PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
        name          : PropTypes.string.isRequired,
        text          : PropTypes.string.isRequired,
        checked       : PropTypes.bool.isRequired,
        defaultChecked: PropTypes.bool.isRequired,
        onClickHandler: PropTypes.func.isRequired,
        reverse       : PropTypes.bool
    };

    constructor(props) {
        super(props);

        this.onClickHandler = this.onClickHandler.bind(this);
    }

    render() {
        return (
            this.props.reverse ?
                <label className="label-checkbox item-content">
                    <input type="radio" name={this.props.name} value={this.props.value} checked={this.props.checked} defaultChecked={this.props.defaultChecked} readOnly/>
                    <div className="item-media" onClick={this.onClickHandler}>
                        <i className="icon icon-form-checkbox"></i>
                    </div>
                    <div className="item-inner">
                        <div className="item-title">{this.props.text}</div>
                    </div>
                </label>
                :
                <label className="label-checkbox item-content">
                    <div className="item-inner">
                        <div className="item-title">{this.props.text}</div>
                    </div>
                    <input type="radio" name={this.props.name} value={this.props.value} checked={this.props.checked} defaultChecked={this.props.defaultChecked} readOnly/>
                    <div className="item-media" onClick={this.onClickHandler}>
                        <i className="icon icon-form-checkbox"></i>
                    </div>
                </label>
        );
    }

    onClickHandler() {
        setTimeout(() => {
            this.props.onClickHandler(this.props.value);
        }, 50);
    }
}
