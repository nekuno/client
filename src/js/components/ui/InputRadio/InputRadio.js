import PropTypes from 'prop-types';
import React, { Component } from 'react';
import RoundedIcon from '../RoundedIcon/RoundedIcon.js';
import styles from './InputRadio.scss';

export default class InputRadio extends Component {

    static propTypes = {
        value         : PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
        text          : PropTypes.string.isRequired,
        checked       : PropTypes.bool.isRequired,
        onClickHandler: PropTypes.func.isRequired,
        reverse       : PropTypes.bool
    };

    constructor(props) {
        super(props);

        this.onClickHandler = this.onClickHandler.bind(this);
    }

    onClickHandler() {
        this.props.onClickHandler(this.props.value);
    }

    render() {
        const {reverse, checked, text} = this.props;
        //TODO: Replace with check/un-check icons
        const icon = checked ? 'minus' : 'plus';

        return (
            reverse ?
                <label className={styles.inputRadio} onClick={this.onClickHandler}>
                    <div className={styles.iconWrapper}>
                        <RoundedIcon icon={icon} size={'small'} />
                    </div>
                    <div className={styles.itemWrapper}>
                        <div className={styles.itemText}>{text}</div>
                    </div>
                </label>
                :
                <label className={styles.inputRadio} onClick={this.onClickHandler}>
                    <div className={styles.itemWrapper}>
                        <div className={styles.itemText}>{text}</div>
                    </div>
                    <div className={styles.iconWrapper}>
                        <RoundedIcon icon={icon} size={'small'} />
                    </div>
                </label>
        );
    }
}
