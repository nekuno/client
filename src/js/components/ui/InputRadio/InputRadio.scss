@import '../../../../scss/_partials/colors';

// TODO: All
.input-radio {
  .item-wrapper {
    .item-text {
      //text-transform: none;
      //white-space: normal;
      //width: 50%;
    }
    &::after {
      content: none;
    }
  }

  .icon-wrapper {

  }

  &::after {
    content: none;
  }
}

//.label-checkbox.active-state {
//  background-color: transparent !important;
//}