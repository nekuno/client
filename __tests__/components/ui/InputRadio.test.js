import React from 'react';
import InputRadio from '../../../src/js/components/ui/InputRadio/InputRadio.js';
import styles from '../../../src/js/components/ui/InputRadio/InputRadio.scss';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Test InputRadio component', () => {
    const mockFn = jest.fn();
    const inputRadio = shallow(<InputRadio value={'lorem'} text={'Lorem'} checked={true} onClickHandler={mockFn}/>);

    it('should be defined', () => {
        expect(inputRadio).toBeDefined();
    });
    it('should render correctly', () => {
        expect(inputRadio).toMatchSnapshot();
    });
    it('InputRadio click event', () => {
        inputRadio.find(`.${styles.inputRadio}`).simulate('click');
        expect(mockFn).toHaveBeenCalledTimes(1);
    });
});
