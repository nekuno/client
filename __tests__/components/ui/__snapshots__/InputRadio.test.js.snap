// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test InputRadio component should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputRadio
    checked={true}
    onClickHandler={[MockFunction]}
    text="Lorem"
    value="lorem"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="itemWrapper"
        >
          <div
            className="itemText"
          >
            Lorem
          </div>
        </div>,
        <div
          className="iconWrapper"
        >
          <RoundedIcon
            icon="minus"
            size="small"
          />
        </div>,
      ],
      "className": "inputRadio",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="itemText"
          >
            Lorem
          </div>,
          "className": "itemWrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem",
            "className": "itemText",
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <RoundedIcon
            icon="minus"
            size="small"
          />,
          "className": "iconWrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "icon": "minus",
            "size": "small",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "label",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="itemWrapper"
          >
            <div
              className="itemText"
            >
              Lorem
            </div>
          </div>,
          <div
            className="iconWrapper"
          >
            <RoundedIcon
              icon="minus"
              size="small"
            />
          </div>,
        ],
        "className": "inputRadio",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="itemText"
            >
              Lorem
            </div>,
            "className": "itemWrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Lorem",
              "className": "itemText",
            },
            "ref": null,
            "rendered": "Lorem",
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <RoundedIcon
              icon="minus"
              size="small"
            />,
            "className": "iconWrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "icon": "minus",
              "size": "small",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "label",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
